### ISSUE TITLE
Repeats

### ISSUE THREAD

**Issue by daveyarwood** [alda-lang/alda#132]
The original idea, proposed by myself in #7: 

> **Repeats**

> Self-explanatory. e.g. `c1(~1)x7` should expand to `c1~1~1~1~1~1~1~1` at compile time.

The ideas involved with repeats will go hand in hand with those of variables (#131).

The syntax of Alda has evolved recently -- new syntax idea for repeats is something like this:

    c1[~1]*7

* Parentheses are now reserved for inline Clojure code. Square brackets represent raw "Alda code" objects, which can be manipulated e.g. with repeats.

* @MillerMark proposed (and I agree) that `*` is a better repeat operator than `x` -- see #131.

Variables can also be repeated:

    threeNotes = c d e
    piano: threeNotes*5

I see no reason why individual notes and octave changes couldn't be repeated too:

    guitar: e8*20 >*3 e

**Comment by daveyarwood** [comment]
@Deveritas proposed an alternate repeat syntax mirroring what repeats look like in sheet music:

    ||: c d e :||

or perhaps:

    |: c d e :|

I could see this being useful as a syntactic sugar for "repeat this phrase N times and insert barlines so that each repetition is a measure," which I think could be nice when we implement sheet music generation. (Perhaps we should consider this idea out-of-scope at this point, as it's not immediately useful until we get to the point where we're generating sheet music.)

**Comment by daveyarwood** [comment]
> ```
c1[~1]*7
```

Think I may need to take this particular use case off the table... I think people\* would expect the bracket syntax to be an "event repeater," not a "text repeater." If it were truly a text repeater, things like `[c d e f g]*2` would lead to unexpected results because they would expand to `c d e f gc d e f g`. It's turning out to be more difficult to implement the "text repeater" way in the parser, anyway.

\*myself included, even -- I made this assumption when I wrote the initial examples for this issue!
